src/main/java/com/realexpayments/hpp/sdk/RealexHpp.java

```diff
@@ -67,15 +67,16 @@ public RealexHpp(String secret) {
 	 * <ul>
 	 * <li>Validates inputs</li>
 	 * <li>Generates defaults for security hash, order ID and time stamp (if required)</li>
-	 * <li>Base64 encodes inputs</li>
+	 * <li>Optional to Base64 encode inputs</li>
 	 * <li>Serialises request object to JSON</li>
 	 * </ul>
 	 * </p>
 	 * 
 	 * @param hppRequest
+	 * @param encoded <code>true</code> if the JSON values should be encoded.
 	 * @return String
 	 */
-	public String requestToJson(HppRequest hppRequest) {
+	public String requestToJson(HppRequest hppRequest, boolean encoded ) {
```

Comment:
Why not make this optional like the PHP SDK change? This change would break compatibility for existing integrations that upgrade.

Reply:
Line 127 below makes it optional. Default is encoded.

Reply:
And null values are handled by @JsonInclude(Include.NON_NULL) in the HppRequest class.

Reply:
Sounds good on both fronts. I missed the other method signature for `requestToJson` as I'm used to seeing the main implementation as defined last.